### ISSUE TITLE
Automatic testing of email 2fa

### ISSUE THREAD

**Issue by philomather** [scheb/two-factor-bundle#271]
**Bundle version**: >=4.7.0
**Symfony version**: 4.4.0

**Description**
Hi, I've been having some trouble testing the email 2fa automatically. Before version 4.7.0, just creating a TwoFactorToken in a test action was enough to make the email send, but since then I'm not catching the email any more.
Do you know what change from that version would affect this, and how I could update the test?

**To Reproduce**
```php
     /**
     *
     * @param Request                  $request
     * @param EventDispatcherInterface $eventDispatcher
     * @param SessionInterface         $session
     * @param TokenStorageInterface    $tokenStorage
     *
     * @return RedirectResponse
     */
    public function loginTestAction(
        Request $request,
        EventDispatcherInterface $eventDispatcher,
        SessionInterface $session,
        TokenStorageInterface $tokenStorage
    ): RedirectResponse
    {
        $user = $this->getDoctrine()->getRepository(User::class)->loadUserByUsername($request->query->get('username'));

        $token = new UsernamePasswordToken($user, null, 'main', $user->getRoles());
        if (\boolval($request->query->get('two_factor', null)) === true)
        {
            $token = new TwoFactorToken($token, null, 'main', ['email']);
        }
        $tokenStorage->setToken($token);

        $session->set('_security_main', \serialize($token));

        // Fire the login event manually
        $event = new InteractiveLoginEvent($request, $token);
        $eventDispatcher->dispatch($event);

        return $this->redirectAfterLogin();
    }
```

**Comment by scheb** [comment]
With v4.7.0 the preparation was patched which was a bit flawed before (#223 & #212).

From that version on, per default a two-factor provider doesn't automatically prepare directly after login, but only when the user lands on the 2fa page. Since you're using the login event `InteractiveLoginEvent`, it doesn't trigger preparation in that version (with the default configuration).

Two ways how to fix this:
1) Configure `prepare_on_login: true` on your firewall to force preparation on the login event
2) Dispatch a `TwoFactorAuthenticationEvent` of type `TwoFactorAuthenticationEvents::FORM` instead, that's guaranteed to trigger preparation.

**Comment by philomather** [comment]
Thanks for answering so quickly!
Dispatching an authentication event worked perfectly.
